File: mycode/catkin_ws/src/my_robot_tutorials/scripts/number_publisher.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int64

if __name__ == '__main__':
	rospy.init_node('number_publisher_py', anonymous=True) 

	pub = rospy.Publisher('/number', Int64, queue_size = 10)

	publish_frequency = rospy.get_param("/number_publish_freq")
	number = rospy.get_param("/number_to_publish")
	rospy.set_param("/another_param","Hello from node")
	rate = rospy.Rate(publish_frequency)

	while not rospy.is_shutdown():
		num = Int64()
		num.data = number
		pub.publish(num)
		rate.sleep()

	rospy.loginfo("Node was stopped")

File: mycode/catkin_ws/src/my_robot_tutorials/src/number_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

int main(int argc, char **argv){
	ros::init(argc,argv,"number_publisher_cpp", ros::init_options::AnonymousName);
	ros::NodeHandle nh;

	ros::Publisher pub = nh.advertise<std_msgs::Int64>("/number", 10);

	double publish_frequency;
	nh.getParam("/number_publish_freq",publish_frequency);
	int number;
	nh.getParam("/number_to_publish",number);
	nh.setParam("/just_another_param","Hello from Cpp node");
	ros::Rate rate(publish_frequency);
	while(ros::ok()){
		std_msgs::Int64 num;
		// num.data = 5000;
		num.data = number;
		pub.publish(num);
		rate.sleep();
	}
}